eth/downloader/downloader.go

```diff
@@ -325,52 +367,50 @@ func (d *Downloader) UnregisterPeer(id string) error {
 	// Unregister the peer from the active peer set and revoke any fetch tasks
 	glog.V(logger.Detail).Infoln("Unregistering peer", id)
 	err = d.peers.Unregister(id)
+
+	defer func() {
+		// If this peer was the master peer, abort sync immediately
+		d.cancelLock.RLock()
+		master := id == d.cancelPeer
+		d.cancelLock.RUnlock()
+
+		if master {
+			d.cancel()
+		}
+	}()
+
 	if err != nil {
-		glog.V(logger.Error).Errorln("Unregister failed:", err)
+		glog.V(logger.Warn).Warnln("Unregister failed:", err)
 		return err
 	}
 	d.queue.Revoke(id)
 
-	// If this peer was the master peer, abort sync immediately
-	d.cancelLock.RLock()
-	master := id == d.cancelPeer
-	d.cancelLock.RUnlock()
-
-	if master {
-		d.cancel()
-	}
 	return nil
 }
 
 // Synchronise tries to sync up our local block chain with a remote peer, both
 // adding various sanity checks as well as wrapping it with various log entries.
-func (d *Downloader) Synchronise(id string, head common.Hash, td *big.Int, mode SyncMode) bool {
+func (d *Downloader) Synchronise(id string, head common.Hash, td *big.Int, mode SyncMode) error {
 	err := d.synchronise(id, head, td, mode)
 	switch err {
 	case nil:
 		glog.V(logger.Core).Infof("Peer %s: sync complete", id)
-		return true
-
 	case errBusy:
 		glog.V(logger.Debug).Warnln("sync busy")
-
-	case errTimeout, errBadPeer, errStallingPeer, errEmptyHashSet,
+	case errTimeout, errBadPeer, errStallingPeer,
 		errEmptyHeaderSet, errPeersUnavailable, errTooOld,
 		errInvalidAncestor, errInvalidChain:
 		glog.V(logger.Core).Warnf("Peer %s: drop: %s", id, err)
 		d.dropPeer(id)
 
-	case errCancelBlockFetch, errCancelHeaderFetch, errCancelBodyFetch, errCancelReceiptFetch, errCancelStateFetch, errCancelHeaderProcessing, errCancelContentProcessing:
```

Comment:
Because this is gone, errors get logged even we requested them. That's an example of why the `ErrWasRequested` convenience function above is nice to have.

Reply:
And even though their log level is just `Core`, I think it's better than just ignoring them.